### ISSUE TITLE
Linux Packages

### ISSUE THREAD

**Issue by trymeouteh** [haveno-dex/haveno#270]
Please consider packaging the Linux client into the following formats...

**AppImage** (Works on any distro, portable, entire app is inside a single file (not a folder), easy to install, latest version available)

**AUR** (Work on arch distros, easy to deploy, safe and secure download, automatic updates, easy to install, latest version available)

**Pacstall** (Work on debian/ubuntu distros, safe and secure download, automatic updates, latest version available)

Not trying to start a "linux package debate/war" but I have used all of the linux packages out there and I do think these three are the best for the follow reasons above compared to other alternatives such as official arch/debian/ubuntu repos, PPAs, tarballs, snaps and flatpaks.

**Comment by MrCyjaneK** [comment]
Can be discussed as a part of #255 I believe.

**Comment by kinghat** [comment]
id say it would benefit the project to pick a linux package that can be distributed through an app store and be distro agnostic. that currently is either flatpak or snaps. imo id take either over any of the alternatives, but if it was down to one or the other, itd be flatpak.

**Comment by k4r4b3y** [comment]
+1 for the AppImage request. I use feather with its appimage releases. Quite easy to work with them. Just download and run, no need to installa thing.

**Comment by woodser** [comment]
Probably support for AppImage or anything else needs to be added to the desktop project's [gradle build](https://github.com/haveno-dex/haveno/blob/master/desktop/package/package.gradle), which packages the application for distribution, in case someone can help with the implementation.

**Comment by k4r4b3y** [comment]
@chaserene is it true that flatpak requires `sudo` for the initial install of the program (let's say, haveno)?  If so, that's a big disadvantage for the users of TailsOS and Qubes-Whonix, as TailsOS is made to not rely on `sudo` as much as possible, while Qubes-Whonix doesn't play comfortably with moving binaries to the root directory (root dirs in QubesOS are read-only and comes from the TemplateVM).

AppImage, on the other hand, works with no `sudo` necessary, at any point.  User just downloads the Haveno.Appimage file and gives it user exec permission and then runs it with `./Haveno.AppImage` command.

**Comment by woodser** [comment]
AppImage would be a nice addition. Maybe worth opening a separate issue and adding a bounty specifically for that.

**Comment by k4r4b3y** [comment]
> Maybe worth opening a separate issue and adding a bounty specifically for that.

I can open a new issue for requesting an AppImage on haveno project's github page, but I don't know how the bounties work here.  Do I pledge some amount of XMR for the successful closure of the github issue that I open?

**Comment by woodser** [comment]
If you open the issue, we can pledge a bounty of 1 XMR, and add to that if there are external donations pledged for it. Please specify any terms if applicable.

**Comment by k4r4b3y** [comment]
@woodser I would open the issue, however, I am not clear on some technical stuff regarding the bounty criteria for payment, such as:

1. if the appimage gets produced by someone else, other than you (woodser), how can I be sure of its authenticity?  How powerful is the appimage packager in potentially spreading malware/malformed software?
2. will the users still be able to compare hashes of the appimage and ascertain its authenticity?

If I get satisfactory answers to such questions, I am willing to pledge +1 XMR on top of your 1 XMR.

**Comment by woodser** [comment]
> if the appimage gets produced by someone else, other than you (woodser), how can I be sure of its authenticity? How powerful is the appimage packager in potentially spreading malware/malformed software?

That's beyond our scope here. What we're paying for with this bounty is the ability for anyone to create an AppImage from the Haveno repo, so any necessary scripts, documentation, or additional support should be added directly to the official Haveno repo. Then third parties running on mainnet (excludes the official repo), can follow those steps to create and distribute an AppImage for their network instance.

> will the users still be able to compare hashes of the appimage and ascertain its authenticity?

They should be able to compare the hash against the AppImage released by the operating network to verify it's the same. If the AppImage is deterministically built, it's possible to verify the expected hash independently as well, but that's not a requirement I don't think.

> If I get satisfactory answers to such questions, I am willing to pledge +1 XMR on top of your 1 XMR.

We typically require people to donate their pledge to the project in order to include it in the advertised bounty, with the option of reimbursement if needed.

**Comment by woodser** [comment]
> @k4r4b3y: how can I be sure of its authenticity?

I forgot to mention that these installers can be built automatically using GitHub Actions, so users can compare the downloadable installers against the build from GitHub Actions to verify their integrity.

**Comment by woodser** [comment]
I've opened a bounty for building AppImage automatically: https://github.com/haveno-dex/haveno/issues/1222

**Comment by k4r4b3y** [comment]
Thanks.  I just sent 1 XMR donation to the HavenoDex donation address [here](https://github.com/haveno-dex/haveno?tab=readme-ov-file#monero).

**Comment by Jabster28** [comment]
@k4r4b3y 
> is it true that flatpak requires `sudo` for the initial install of the program

Nope, you need sudo to install Flatpak (the program) afaik but you [can install individual programs with the `--user` tag](https://docs.flatpak.org/en/latest/using-flatpak.html#system-versus-user) and you get all of the same guarantees. See https://docs.flatpak.org/en/latest/introduction.html#reasons-to-use-flatpak :

> **Rootless install**: elevated privileges are not required when installing a Flatpak application or a runtime.

**Comment by Jabster28** [comment]
Also  @trymeouteh  for an AUR package, shouldn't that be done outside of the repository? Or are you suggesting that a sample `PKGBUILD` is made in the repo?

**Comment by Jabster28** [comment]
@k4r4b3y 

> if the appimage gets produced by someone else, other than you (woodser), how can I be sure of its authenticity? How powerful is the appimage packager in potentially spreading malware/malformed software?

AppImages support [signing](https://docs.appimage.org/packaging-guide/optional/signatures.html) with GPG keys (checking the signatures complex though, you might be better off just distributing it with a `.sig`), but if you're referring to different networks releasing their own AppImages, the binaries have a different signature from woodser's and thus the network admins would need to use their own keys to sign it. That would still let you confirm that it hasn't been edited by anyone aside from the network admins, however.

**Comment by woodser** [comment]
AppImage is now supported, along with flatpak. I think we can close this issue and re-open for other specific requests.
